### ISSUE TITLE
Python 3.12: no pkg_resources

### ISSUE THREAD

**Issue by eccles** [fubuloubu/py-trie#142]
### What happened?

I added support for python 3.12 in this PR: https://github.com/rkvst/rkvst-receipt-scitt/pull/11.

The CICDaction failes because pkg_resources no longer exists.

See https://github.com/rkvst/rkvst-receipt-scitt/actions/runs/6559511687/job/17815261244?pr=11

### Code that produced the error

```python
File "/opt/hostedtoolcache/Python/3.12.0/x64/lib/python3.12/site-packages/trie/__init__.py", line 1, in <module>
    import pkg_resources
ModuleNotFoundError: No module named 'pkg_resources'
```

### Fill this section in if you know how this could or should be fixed

Use importlib.resources instead.

https://importlib-resources.readthedocs.io/en/latest/migration.html

### py-trie Version

2.1.0

### Python Version

3.12

### Operating System

linux

### Output from `pip freeze`

```shell
Run python3 -m pip freeze # required to report issue to py-trie
alabaster==0.7.13
asn1crypto==1.5.1
astroid==3.0.1
attrs==23.1.0
autopep8==2.0.4
Babel==2.13.0
black==23.10.0
boolean.py==4.0
build==1.0.3
CacheControl==0.13.1
cbor2==5.4.6
certifi==2023.7.22
certvalidator==0.11.1
cffi==1.16.0
charset-normalizer==3.3.0
click==8.1.7
coverage==7.3.2
cryptography==41.0.4
cyclonedx-python-lib==4.2.3
cytoolz==0.12.2
defusedxml==0.7.1
dill==0.3.7
docutils==0.18.1
ecdsa==0.18.0
eth-hash==0.5.2
eth-typing==3.5.0
eth-utils==2.2.2
filelock==3.12.4
hexbytes==0.3.1
html5lib==1.1
idna==3.4
imagesize==1.4.1
importlib-metadata==6.8.0
isort==5.12.0
jaraco.classes==3.3.0
jeepney==0.8.0
Jinja2==3.1.2
keyring==24.2.0
license-expression==30.1.1
lxml==4.9.3
markdown-it-py==2.2.0
MarkupSafe==2.1.3
mccabe==0.7.0
mdit-py-plugins==0.3.5
mdurl==0.1.2
mistune==2.0.5
more-itertools==10.1.0
msgpack==1.0.7
mypy-extensions==1.0.0
myst-parser==1.0.0
nh3==0.2.14
oscrypto==1.3.0
packageurl-python==0.11.2
packaging==23.2
pathspec==0.11.2
pip-api==0.0.30
pip-requirements-parser==32.0.1
pip_audit==2.6.1
pkginfo==1.9.6
platformdirs==3.11.0
py-serializable==0.11.1
pycodestyle==2.11.1
pycose==1.0.1
pycparser==2.21
pycryptodome==3.19.0
pyenchant==3.2.2
Pygments==2.16.1
pylint==3.0.1
pyparsing==3.1.1
pyproject_hooks==1.0.0
PyYAML==6.0.1
readme-renderer==42.0
requests==2.31.0
requests-toolbelt==1.0.0
rfc3339==6.2
rfc3986==2.0.0
rich==13.6.0
rlp==3.0.0
SecretStorage==3.3.3
six==1.16.0
snowballstemmer==2.2.0
sortedcontainers==2.4.0
Sphinx==5.3.0
sphinx-gallery==0.12.2
sphinx-rtd-theme==1.2.2
sphinx_mdinclude==0.5.3
sphinxcontrib-applehelp==1.0.7
sphinxcontrib-devhelp==1.0.5
sphinxcontrib-htmlhelp==2.0.4
sphinxcontrib-jquery==4.1
sphinxcontrib-jsmath==1.0.1
sphinxcontrib-qthelp==1.0.6
sphinxcontrib-serializinghtml==1.1.9
sphinxcontrib-spelling==8.0.0
toml==0.10.2
tomlkit==0.12.1
toolz==0.12.0
trie==2.1.1
twine==4.0.2
typing_extensions==4.8.0
urllib3==2.0.7
webencodings==0.5.1
xq==0.0.4
zipp==3.17.0
```

**Comment by antazoey** [comment]
Ape is hitting this too

**Comment by pacrob** [comment]
The reference to `pkg_resources` was removed in `v3.0.0`, and official python 3.12 support just merged in #154. It should be released early next week.

**Comment by antazoey** [comment]
Yeah I realized I was on 2.2.0, which is installable on 3.12 even though it seems to have issues like this.
